src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+          nextDate = date.getDate() + 1;
+          monthToReturn = date.getMonth();
+          yearToReturn = date.getFullYear();
+        }
+      } else if (!is30DayMonth) {
+        if (date.getDate() === 31) {
+          nextDate = 1;
+          monthToReturn = (date.getMonth() + 1) % 12;
+          yearToReturn = date.getFullYear();
+        } else {
+          nextDate = date.getDate() + 1;
+          monthToReturn = date.getMonth();
+          yearToReturn = date.getFullYear();
+        }
+      }
+
+      return {
+        date: nextDate,
+        month: monthToReturn,
+        year: yearToReturn,
+      };
+    };
+
+    const disableDisallowedDays = date => {
+      let dayIn = selectedDateIn;
+      let nextDate = getNextDate(dayIn);
+      let shouldDisableDate = !(
+        (date.getDate() === dayIn.getDate() &&
+          date.getMonth() === dayIn.getMonth() &&
+          date.getYear() === dayIn.getYear()) ||
+        (date.getDate() === nextDate.date &&
+          date.getMonth() === nextDate.month &&
+          date.getFullYear() === nextDate.year)
+      );
+      return shouldDisableDate;
+    };
+
+    const handleCloseSnackbar = (event, reason) => {
+      if (reason === 'clickaway') {
+        return;
+      }
+
+      setSnackbarOpen(false);
+    };
+
+    const jobDropdown = (
+      <FormControl
+        disabled={userJobs === null || userJobs.length === 0}
+        style={{
+          width: 252,
+        }}
+      >
+        <InputLabel>Jobs</InputLabel>
+        <Select
+          value={selectedJob}
+          onChange={e => {
+            setSelectedJob(e.target.value);
+          }}
+          input={<Input id="job" />}
+        >
+          <MenuItem label="None" value="">
+            <em>None</em>
+          </MenuItem>
+          {jobsMenuItems}
+        </Select>
+      </FormControl>
+    );
+
+    let errorText;
+    if (enteredFutureTime) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot begin or end in the future.
+      </Typography>
+      );
+    } else if (timeOutIsBeforeTimeIn) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot end before it starts.
+      </Typography>
+      );
+    } else if (shiftTooLong) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot be longer than 20 hours.
+      </Typography>
+      );
+    } else if (shiftTooShort) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          A shift cannot be shorter than 15 minutes.
+      </Typography>
+      );
+    } else if (isOverlappingShift) {
+      errorText = (
+        <Typography variant="overline" color="error">
+          The entered shift conflicts with a previous shift.
+      </Typography>
+      );
+    }
+    else {
+      errorText = <></>;
+    }
+
+    const onDatetimeSelectorClose = () => {
+      setIsOverlappingShift(false);
```

Comment:
Why is overlapping shift set false here (as opposed to somewhere else)?

Reply:
My thinking was that if a user ever sees the associated error, after they make a change to their timein/timeout the error should go away. Now that you mention it, that sounds like an operation that should go in one of the `onChange` functions instead of this one.

Reply:
Yeah, maybe in the same place where you set it true, you also set it false- under the right conditions, of course.

Reply:
The issue is that it's set to true in response to an http fetch. When it's set, we can't run another fetch and try to set it to false because to do that we'd not to save again, but that's disabled when it's set to true

Reply:
Oh, I see. That's difficult. . .